crates/common/validator/src/sync_committee.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::HashSet;
+
+use anyhow::{anyhow, ensure};
+use ream_consensus::{
+    constants::{EPOCHS_PER_SYNC_COMMITTEE_PERIOD, SYNC_COMMITTEE_SIZE},
+    electra::beacon_state::BeaconState,
+    misc::compute_epoch_at_slot,
+};
+
+use crate::constants::SYNC_COMMITTEE_SUBNET_COUNT;
+
+pub fn compute_sync_committee_period(epoch: u64) -> anyhow::Result<u64> {
+    Ok(epoch / EPOCHS_PER_SYNC_COMMITTEE_PERIOD)
+}
+
+pub fn is_assigned_to_sync_committee(
+    state: &BeaconState,
+    epoch: u64,
+    validator_index: u64,
+) -> anyhow::Result<bool> {
+    let sync_committee_period = compute_sync_committee_period(epoch)?;
+    let current_epoch = state.get_current_epoch();
+    let current_sync_committee_period = compute_sync_committee_period(current_epoch)?;
+    let next_sync_committee_period = current_sync_committee_period + 1;
+    ensure!(
+        sync_committee_period == current_sync_committee_period
+            || sync_committee_period == next_sync_committee_period,
+        "Validator is not assigned to sync committee for period {sync_committee_period} (current: {current_sync_committee_period}, next: {next_sync_committee_period})"
+    );
+
+    let validator_pubkey = state
+        .validators
+        .get(validator_index as usize)
+        .ok_or_else(|| anyhow!("Validator index out of bounds"))?
+        .pubkey
+        .clone();
+
+    if sync_committee_period == current_sync_committee_period {
+        Ok(state
+            .current_sync_committee
+            .pubkeys
+            .contains(&validator_pubkey))
+    } else {
+        Ok(state
+            .next_sync_committee
+            .pubkeys
+            .contains(&validator_pubkey))
+    }
+}
+
+pub fn compute_subnets_for_sync_committee(
+    state: &BeaconState,
+    validator_index: u64,
+) -> anyhow::Result<HashSet<u8>> {
```

Comment:
```suggestion
) -> anyhow::Result<HashSet<u64>> {
```

As stated above SubnetID is a u64